app/controllers/nonsipp/employercontributions/EmployerContributionsMemberListController.scala

```diff
@@ -75,45 +78,42 @@ class EmployerContributionsMemberListController @Inject()(
 
   private def onPageLoadCommon(srn: Srn, page: Int, mode: Mode, showBackLink: Boolean = true)(
     implicit request: DataRequest[AnyContent]
-  ): Result = {
-    val optionList: List[Option[NameDOB]] = request.userAnswers.membersOptionList(srn)
-    if (optionList.flatten.nonEmpty) {
-      optionList
-        .zipWithRefinedIndex[Max300.Refined]
-        .map { indexes =>
-          val employerContributions = buildEmployerContributions(srn, indexes)
-          val noPageEnabled = !request.userAnswers.get(EmployerContributionsPage(srn)).getOrElse(false)
-          Ok(
-            view(
-              viewModel(
-                srn,
-                page,
-                mode,
-                employerContributions,
-                viewOnlyUpdated = if (mode == ViewOnlyMode && request.previousUserAnswers.nonEmpty) {
-                  getCompletedOrUpdatedTaskListStatus(
-                    request.userAnswers,
-                    request.previousUserAnswers.get,
-                    pages.nonsipp.employercontributions.Paths.memberEmpContribution
-                  ) == Updated
-                } else {
-                  false
-                },
-                optYear = request.year,
-                optCurrentVersion = request.currentVersion,
-                optPreviousVersion = request.previousVersion,
-                compilationOrSubmissionDate = request.userAnswers.get(CompilationOrSubmissionDatePage(srn)),
-                noPageEnabled,
-                showBackLink = showBackLink
-              )
+  ): Result =
+    request.userAnswers.completedMembersDetails(srn) match {
+      case Left(err) =>
+        logger.warn(s"Error when fetching completed member details - $err")
+        Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
+      case Right(Nil) =>
+        logger.warn(s"No completed member details for srn $srn")
+        Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
```

Comment:
Rather than suppress issues like the old impl, we now log any errors when fetching completed members. I suppose this could be rolled into a helper function in the future